mlrun/datastore/sources.py

```diff
@@ -826,6 +875,252 @@ def add_nuclio_trigger(self, function):
+        )
+
+    def is_iterator(self):
+        return True if self.attributes.get("chunksize") else False
+
+
+class SqlDBSource(BaseSourceDriver):
+
+    kind = "sqldb"
+    support_storey = True
+    support_spark = False
+    _SQL_DB_PATH_STRING_ENV_VAR = "SQL_DB_PATH_STRING"
+
+    def __init__(
+        self,
+        name: str = "",
+        max_results_for_table: int = None,
+        chunksize: int = None,
+        key_field: str = None,
+        time_field: str = None,
+        schedule: str = None,
+        start_time=None,
+        end_time=None,
+        db_path: str = None,
+        collection_name: str = None,
+        spark_options: dict = None,
+    ):
+        """
+        Reads SqlDB as input source for a flow.
+
+        example::
+            db_path = "sqlite:///stockmarket.db"
+            source = SqlDBSource(
+                collection_name='source_name', db_path=self.db, key_field='key'
+            )
+
+        :parameter name:            source name
+        :parameter chunksize:       number of rows per chunk (default large single chunk)
+        :parameter key_field:       the column to be used as the key for the collection.
+        :parameter time_field:      the column to be parsed as the timestamp for events. Defaults to None
+        :parameter start_time:      filters out data before this time
+        :parameter end_time:        filters out data after this time
+        :parameter schedule:        string to configure scheduling of the ingestion job. For example '*/30 * * * *' will
+                                    cause the job to run every 30 minutes
+        :param db_path:             url string connection to sql database.
+                                    If not set, the SQL_DB_PATH_STRING environment variable will be used.
+        :parameter collection_name: the name of the collection to access,
+                                    from the current database
+        :parameter spark_options:   additional spark read options
+        """
+
+        db_path = db_path or os.getenv(self._SQL_DB_PATH_STRING_ENV_VAR)
+        if db_path is None:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"cannot specify without db_path arg or secret {self._SQL_DB_PATH_STRING_ENV_VAR}"
+            )
+        attrs = {
+            "max_results": max_results_for_table,
+            "chunksize": chunksize,
+            "spark_options": spark_options,
+            "collection_name": collection_name,
+            "db_path": db_path,
+        }
+        attrs = {key: value for key, value in attrs.items() if value is not None}
+        super().__init__(
+            name,
+            attributes=attrs,
+            key_field=key_field,
+            time_field=time_field,
+            schedule=schedule,
+            start_time=start_time,
+            end_time=end_time,
+        )
+
+    def to_dataframe(self):
+        import sqlalchemy as db
+
+        query = self.attributes.get("query")
```

Comment:
if self.attributes is a dictionary, it will make more sense to use square brackets. (brackets will work even for defaultdict - a dictionary that will return automatically a default value for missing parameters).